### PR TITLE
Move filter into it's own model; default to homeless types

### PR DESCRIPTION
**Pull request by eanders**
@ttoomey, is this roughly the change you need for background PDF processing?
@awisema, can you give this the usual QA pass?

### CODE REVIEW

**Review by awisema — APPROVED** [review]

File: app/models/filters/performance_dashboard.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+      new_first = last
+      self.end = first
+      self.start = new_first
+    end
+
+    def ensure_date_span
+      return if last - first < 365
+
+      self.end = first + 1.years
```
Comment (awisema) [inline app/models/filters/performance_dashboard.rb]: `self.end = first + 1.year - 1.day` to be consistent with the default range?

**Review by awisema — CHANGES_REQUESTED** [review]
Maybe hide the CoC Code from the filter summary if it is hidden on the projects and funding tab?

File: app/models/filters/performance_dashboard.rb (hunk 1 above)
Comment (eanders) [reply app/models/filters/performance_dashboard.rb]: Yep

**Review by ttoomey — COMMENTED** [review]
@eanders code looks good to me, I can't speak to the domain since I'm too new.

The only feedback I have is that using `end` as a variable could be a headache since it's a reserved word.

**Comment by eanders** [comment]
> @eanders code looks good to me, I can't speak to the domain since I'm too new.
> 
> The only feedback I have is that using `end` as a variable could be a headache since it's a reserved word.

Yeah, we've seen that for sure.  That's a legacy now that we should tear out, but since it's a reserved word and spread around quite liberally, it's a bit of a lift to remove.

**Pull request merged**